e2e/pushpull_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/docker/app/internal"
+	"gotest.tools/assert"
+	"gotest.tools/assert/cmp"
+	"gotest.tools/fs"
+	"gotest.tools/icmd"
+)
+
+type dindSwarmAndRegistryInfo struct {
+	swarmAddress    string
+	registryAddress string
+	configuredCmd   icmd.Cmd
+	stopRegistry    func()
+}
+
+func runWithDindSwarmAndRegistry(t *testing.T, todo func(dindSwarmAndRegistryInfo)) {
```

Comment:
missing `t.Helper()` I think?

Reply:
I don't really think so, as this will run the todo func and it might then be difficult to track failures happening in the callback...

Reply:
I see, yes, that makes sense, adding `t.Helper()` would put the blame on the line calling this function, not the correct line in the multiline lambda function used as an argument to it.